### PR TITLE
Exclude Squiz.Commenting.FunctionComment.ThrowsNotCapital sniff

### PR DESCRIPTION
**Pull request by deeky666**
I suggest allowing descriptions for `@throws` to be started with a non-capital letter as it is more natural to compose a complete sentence out of the line like:

```
 @throws RuntimeException if the file could not be written.
```

As a further improvement one might even enforce sentences like this (starting with `if`).
What do you guys think?

### CODE REVIEW

File: tests/input/example-class.php [hunk 1]
```diff
@@ -61,6 +61,9 @@ public function isBaz() : bool
         return $this->baz;
     }
 
+    /**
+     * @throws \InvalidArgumentException if this example cannot baz.
```
Comment (lcobucci) [inline tests/input/example-class.php]: Shouldn't be this instead (otherwise we're mixing sniffs fixes)?

```diff
- @throws \InvalidArgumentException if this example cannot baz.
+ @throws InvalidArgumentException If this example cannot baz.
```

Comment (deeky666) [reply tests/input/example-class.php]: Hmmm yes but it should be `@throws InvalidArgumentException if this example cannot baz.` then so that we can test non-capital won't trigger an error.

Comment (lcobucci) [reply tests/input/example-class.php]: @deeky666 this file represents the input of the test, so IMO if we want to standardise things I'd expect to have a sniff that fixes things automatically (so the `input` would contain things in capital which would be fixed in the `output`).

**Comment by deeky666** [comment]
@lcobucci can you give an example of a sentence you would put there? Because IMO `@throws` tag descriptions are used to tell the user under which circumstances that particular exception is thrown, thus `if` is used anyways most of the time. As a positive side effect you don't have to duplicate words like "throws" which saves you some words and leaves you more space in the line.

File: tests/input/example-class.php (hunk 1 above)
Comment (deeky666) [reply tests/input/example-class.php]: @lcobucci I get that but there isn't really anything to fix here because we do not explicitly disallow capital letters. How can we test then, that this sniff is excluded now?

**Comment by kukulich** [comment]
@deeky666 You should throw more specific exception with more specific name so you don’t have to write the description at all

**Comment by deeky666** [comment]
@kukulich that is true but is not really the topic of this RFC. There will always be situation where you might not be able to throw a distinct exception for every type of error. Imagine file operations, where you cannot safely say what exactly went wrong when trying to write a file. You can check some preconditions beforehand but that does not prevent you from running into race conditions. This is just one example. In a perfect world I would 100% agree with you. Also we are not explicitly disallowing descriptions for `@throws` do this RFC is perfectly valid IMO.

**Comment by deeky666** [comment]
Descriptions for thrown exceptions especially are useful in interfaces where you might not know the technical details of when to throw which exception during implementation.

**Comment by lcobucci** [comment]
@deeky666 https://github.com/doctrine/doctrine2/blob/0b7d878cd340fed70e22404daa9656bb06cec74a/lib/Doctrine/ORM/Cache/Region.php#L35 surely we could rephrase it to have `If` or `When`.

**Comment by deeky666** [comment]
@lcobucci I see but as a fun fact this contradicts to https://github.com/doctrine/doctrine2/blob/0b7d878cd340fed70e22404daa9656bb06cec74a/lib/Doctrine/ORM/Cache/Region.php#L24 where it is a similar situation :D

File: tests/input/example-class.php (hunk 1 above)
Comment (lcobucci) [reply tests/input/example-class.php]: @deeky666 so the question is: do we want to have it standardised or not? If yes, we should have a sniff and create the correct expectations.

We should have had tests to cover the current behaviour, though.

**Comment by lcobucci** [comment]
@deeky666 we're quite far from perfection 😂

**Comment by Ocramius** [comment]
This patch is a clear :+1: from my side for a few reasons:

 1. Last time I generated API docs, I was still in high school
 2. I don't rely on API descriptions in the IDE either: I jump to the method definitions and read the docblock (I do the same in Java btw)
 3. `@throws AnApple when gravity is to be discovered` is perfectly fine, and just reads awkwardly when  having to restart a sentence

**Comment by carusogabriel** [comment]
Can we vote for this one?

/cc @Ocramius @Majkl578 @lcobucci @jwage @morozov @malarzm @doctrine/team-coding-standards

**Comment by Majkl578** [comment]
I already voted against.

**Comment by morozov** [comment]
The RFC looks like a matter of personal taste to me. Depending on whether we want to use the exception descriptions somewhere else than the source code, the way how we interpret the original annotation may be different.

The `@throws if […]` construct reads naturally but looks inconsistent with `@params`, `@return` and maybe some others. Besides, according to https://github.com/doctrine/coding-standard/pull/52#pullrequestreview-109502951, it doesn't have to read as a sentence.

**Comment by Ocramius** [comment]
Note: had to dismiss the 3 :-1: reviews in order to allow merg: unsure why github started behaving like that

**Comment by Majkl578** [comment]
It was 3 👎 + one rather -1-ish review from @morozov.

I would prefer not merging such controversial RFCs.

**Comment by Ocramius** [comment]
That's what the voting is for: if we introduce an even tougher process for this we'll just get into gridlock. You can always open a new PR to revert this one.
